File: src/leetcode/intersection-of-multiple-arrays/intersection-of-multiple-arrays.js
/**
 * Leetcode #2248
 * Difficulty: Easy
 * https://leetcode.com/problems/intersection-of-multiple-arrays/
 */
/**
 * @param {number[][]} nums
 * @return {number[]}
 */
export const intersection = nums => {
  let set = new Set(nums[0]);

  const filterSet = num => set.has(num);

  for (let i = 1; i < nums.length; i += 1) {
    set = new Set(nums[i].filter(filterSet));
  }

  return Array.from(set).sort((a, b) => a - b);
};

File: src/leetcode/intersection-of-multiple-arrays/intersection-of-multiple-arrays.rs
use std::collections::HashSet;
use std::iter::FromIterator;

impl Solution {
    pub fn intersection(nums: Vec<Vec<i32>>) -> Vec<i32> {
        let mut set = HashSet::<i32>::from_iter(nums[0].iter().cloned());

        for i in 1..nums.len() {
            set = HashSet::<i32>::from_iter(
                nums[i]
                    .iter()
                    .filter(|num| set.contains(&num))
                    .cloned()
            );
        }

        let mut ans = Vec::from_iter(set);
        ans.sort();
        ans
    }
}
